scenario/state.py

```diff
@@ -48,6 +60,26 @@
 }
 
 
+def is_relation_event(name: str) -> bool:
+    """Whether the event name indicates that this is a relation event."""
+    return any(map(name.replace("_", "-").endswith, RELATION_EVENTS_SUFFIX))
```

Comment:
Would this check need to apply to custom events? Custom events may end with any of the suffixes above.

Reply:
Very good point. I suspect you'd need to disable the consistency checks in that case.
Crucially this is only an issue if you attempt to emit the custom event directly, instead of going through the 'proper' channel of triggering whatever event causes that custom event to be emitted in turn.

Reply:
Still debating with myself if we should allow that pattern in the first place. Will put a bunch of warnings/todo's on that, but it's an edge case.

Reply:
TLDR: if you have an endpoint called 'foo' and you choose to call a custom event `foo-relation-changed`, then it will show up as inconsistent. False negative we can do nothing about for now except disable the checker.

Reply:
Final thought:
I think the 'I am trying to emit a foo-relation-changed and I forgot to add foo to my metadata OR to my state.relations' situation is more common than 'I created a custom event and I called it exactly like a builtin relation, workload, or storage event'.
Therefore, we'll guard against the first case and if you're N.2 you'll have to disable the checker, for now. If you like to keep thinking about it we can elaborate in a new issue, we can change or revert this very easily.